### ISSUE TITLE
issue in mask indexing

### ISSUE THREAD

**Issue by stat17-hb** [scikit-learn-contrib/skope-rules#38]
Hi, thank you for sharing this great package.

However, I think I might find a mistake in the mask indexing.

`mask = ~samples`

samples is numpy array, and when you put ~, you can get -(value+1).

ex.
samples = np.array([1,2,3,4])
~samples
[-2, -3, -4, -5]

please check this issue.

Thanks!

**Comment by ngoix** [comment]
thanks for noticing this, which happened after this change on sklearn https://github.com/scikit-learn/scikit-learn/pull/9723
